src/Sulu/Component/Content/SmartContent/ContentDataProvider.php

```diff
@@ -262,31 +323,60 @@ private function load(array $options, $limit)
      * @param array $data
      * @param string $locale
      *
-     * @return array
+     * @return ContentDataItem[]
+     */
+    private function decorateDataItems(array $data, $locale)
+    {
+        return array_map(
+            function ($item) use ($locale) {
+                return new ContentDataItem($item, $this->getResource($item['uuid'], $locale));
+            },
+            $data
+        );
+    }
+
+    /**
+     * Decorates result with item class.
+     *
+     * @param array $data
+     * @param string $locale
+     *
+     * @return ArrayAccessItem[]
      */
-    private function decorate(array $data, $locale)
+    private function decorateResourceItems(array $data, $locale)
     {
         return array_map(
             function ($item) use ($locale) {
-                $resource = $this->proxyFactory->createProxy(
-                    PageDocument::class,
-                    function (
-                        &$wrappedObject,
-                        LazyLoadingInterface $proxy,
-                        $method,
-                        array $parameters,
-                        &$initializer
-                    ) use ($item, $locale) {
-                        $initializer = null;
-                        $wrappedObject = $this->documentManager->find($item['uuid'], $locale);
-
-                        return true;
-                    }
-                );
-
-                return new ContentDataItem($item, $resource);
+                return new ArrayAccessItem($item['uuid'], $item, $this->getResource($item['uuid'], $locale));
             },
             $data
         );
     }
+
+    /**
+     * Returns Proxy Document for uuid.
+     *
+     * @param string $uuid
+     * @param string $locale
+     *
+     * @return object
+     */
+    private function getResource($uuid, $locale)
+    {
+        return $this->proxyFactory->createProxy(
+            PageDocument::class,
+            function (
+                &$wrappedObject,
+                LazyLoadingInterface $proxy,
+                $method,
+                array $parameters,
+                &$initializer
+            ) use ($uuid, $locale) {
+                $initializer = null;
+                $wrappedObject = $this->documentManager->find($uuid, $locale);
+
+                return true;
+            }
+        );
```

Comment:
Of what purpose is the proxy here? The entire document is already loaded in the closure, isn't it?